### DISCUSSION TITLE
Export a component as an alias

### DISCUSSION THREAD

**Discussion by xypnox** [fastn-stack/fastn#1795] [Ideas & RFCs]
Export a component declared inside a file or imported from some other dependency with a different alias.

```
-- component A:
-- end: A

-- export alias B: A  
```

## Usecases

Re-expose larger components as defaults or rename a component to a smaller version when used externally.

Ex: There could be several hero components in a library, and `hero-image-right-cta` might be exported as `hero` for ease of use as it would be the most used variant.
Similarly, `secondary-button` could be exported as `button`.

**Comment by amitu** [comment]
1. We are using `as <alias>` in a few places for aliasing, so I would prefer that. Also I am not sure we need `export` to export, everything in a module are automatically exported. 

If we are exporting symbol defined in another module and want alias feature, we should do this this:

```ftd
-- import: foo
export: bar as baz
```

Because doing so without alias is currently is done like this:

```ftd
-- import: foo
export: bar
```

So this is logical change. 

For symbols defined in current file, we can use an alias type:

```
-- alias foo: bar
```

This can be generic alias for anything, records, variables, components. Or we do:

```
-- alias: foo as bar, x as y
```

This allows more than one aliases to be defined in one line (not sure I am fond of that, as that does not scale to many aliases, unless we allow:

```
-- alias: (
    foo as bar, 
    x as y
)
```

Not sure I like it either.
